org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/mock/RegisterBeanTestRule.java

```diff
@@ -38,8 +39,11 @@ public RegisterBeanTestRule(Class<? super BEAN> beanClazz, Supplier<BEAN> mockSu
   }
 
   public void registerBean() {
+    // load existing bean (optional) to overwrite potential replacements of m_beanClazz (use actual registered beanClazz)
+    // this may even result in an error if our mock does not implement the currently registered beanClazz, it is necessary however as otherwise it would not be guaranteed that our mock would be used
```

Comment:
should we add an assertion for this in order to facilitate finding errors in tests?